raiden-ts/src/transfers/epics/init.ts

```diff
@@ -168,6 +177,65 @@ export function initQueuePendingReceivedEpic(
   );
 }
 
+/**
+ * @param action$ - Observable of unresolved transfer.request actions
+ * @param state$ - Observable of RaidenStates
+ * @param deps - Epics dependenceis
+ * @param deps.config$ - Config observable
+ * @returns Observable of pathFind.request and resolved transfer.request actions
+ */
+export function transferRequestResolveEpic(
+  action$: Observable<RaidenAction>,
+  {}: Observable<RaidenState>,
+  { config$ }: RaidenEpicDeps,
+) {
+  return action$.pipe(
+    dispatchRequestAndGetResponse(pathFind, (dispatch) =>
+      action$.pipe(
+        filter(transfer.request.is),
+        filter(
+          (action): action is transfer.request & { payload: { resolved: false } } =>
+            !action.payload.resolved,
+        ),
+        mergeMap((action) =>
+          dispatch(
+            pathFind.request(
+              pick(action.payload, ['paths', 'pfs'] as const),
+              pick(action.payload, ['tokenNetwork', 'target', 'value'] as const),
+            ),
+          ).pipe(
+            withLatestFrom(
+              action$.pipe(
+                filter(matrixPresence.success.is),
+                filter((a) => a.meta.address === action.payload.target),
+              ),
+              config$,
+            ),
+            map(([route, targetPresence, { encryptSecret }]) => {
+              const resolved =
+                (action.payload.encryptSecret ?? encryptSecret) && action.payload.secret
+                  ? metadataFromPaths(route.payload.paths, targetPresence, {
+                      secret: action.payload.secret,
+                      amount: action.payload.value,
+                      payment_identifier: action.payload.paymentId,
+                    })
+                  : metadataFromPaths(route.payload.paths, targetPresence);
+              return transfer.request(
+                {
+                  ...omit(action.payload, ['paths', 'pfs', 'encryptSecret'] as const),
+                  ...resolved,
+                },
+                action.meta,
+              );
```

Comment:
Usually because of type-safety/narrowing constraints. I'll check your suggestions, thank you. I agree this can be simplified, but our code styles usually vary quite a lot on this regard.

Reply:
True. Check if you like it or not. But in general I think there are some mathematically arithmetic you can use to determine a codes complexity. They are quite objective. I think they definitely have some value. "Clean Code: A Handbook of Agile Software Craftsmanship" - Robert C. Martin is always a great source of inspiration how far you can drive it with clean code. For me this is in first place about readability. The original code just automatically makes me skip parts of your code. That happens to me all the time I'm reading your code...

Reply:
Your solution was not typesafe, because of the null check in the `action.payload.secret`, but I tried to adapt it as well as I could and implement your suggestions. Thanks :heavy_check_mark: